fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java

```diff
@@ -634,6 +641,40 @@ public void initFuzzyModeVariables() {
         this.enableShareHashTableForBroadcastJoin = random.nextBoolean();
     }
 
+    /**
+     * syntax:
+     * all -> use all event
+     * all except event_1, event_2, ..., event_n -> use all events excluding the event_1~n
+     * event_1, event_2, ..., event_n -> use event_1~n
+     */
+    @VariableMgr.VarAttr(name = NEREIDS_EVENT_MODE)
+    public String nereidsEventMode = "all";
+
+    private Set<Class<? extends Event>> parsedNereidsEventMode = EventSwitchParser.parse(Lists.newArrayList("all"));
+
+    public void setEnableNereidsTrace(boolean enableNereidsTrace) {
+        this.enableNereidsTrace = enableNereidsTrace;
+    }
+
+    public String getNereidsEventMode() {
+        return nereidsEventMode;
+    }
+
+    public void setNereidsEventMode(String nereidsEventMode) throws UnsupportedOperationException {
```

Comment:
add check to this function is useless. Doris use reflection to set session vairable value. do not call setXX at all